File: CS/dxSample/Custom Combo Box/CustomRIComboBoxEdit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;

namespace dxSample {
    public class CustomRIComboBox : RepositoryItemComboBox {
        public const string EditorName = "CustomComboBoxEdit";
        public static void Register() {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName,
                   typeof(CustomComboBoxEdit),
                      typeof(CustomRIComboBox),
                      typeof(DevExpress.XtraEditors.ViewInfo.ComboBoxViewInfo),
                      new DevExpress.XtraEditors.Drawing.ButtonEditPainter(),
                      true, EditImageIndexes.ComboBoxEdit));
        }
        public CustomRIComboBox() {
        }
        static CustomRIComboBox() {
            Register();
        }
        public override DevExpress.XtraEditors.Controls.TextEditStyles TextEditStyle {
            get {
                return DevExpress.XtraEditors.Controls.TextEditStyles.DisableTextEditor;
            }
        }
    }
}

File: VB/dxSample/Custom Combo Box/CustomRIComboBoxEdit.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository

Namespace dxSample
	Public Class CustomRIComboBox
		Inherits RepositoryItemComboBox

		Public Const EditorName As String = "CustomComboBoxEdit"
		Public Shared Sub Register()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(CustomComboBoxEdit), GetType(CustomRIComboBox), GetType(DevExpress.XtraEditors.ViewInfo.ComboBoxViewInfo), New DevExpress.XtraEditors.Drawing.ButtonEditPainter(), True, EditImageIndexes.ComboBoxEdit))
		End Sub
		Public Sub New()
		End Sub
		Shared Sub New()
			Register()
		End Sub
        Public Overrides Property TextEditStyle As DevExpress.XtraEditors.Controls.TextEditStyles
            Get
                Return DevExpress.XtraEditors.Controls.TextEditStyles.DisableTextEditor
            End Get
            Set(value As DevExpress.XtraEditors.Controls.TextEditStyles)
                MyBase.TextEditStyle = value
            End Set
        End Property
	End Class
End Namespace
